### DISCUSSION TITLE
error msg="migrate: Error 1146

### DISCUSSION THREAD

**Discussion by Jobsays** [photoprism/photoprism#3833] [Q&A] [answered]
First off, thanks, @lastzero for helping me realize that I can't read on this issue: 
https://github.com/photoprism/photoprism/issues/3199
You're right of course that the container starts despite that error, I just stopped reading at the first error I saw in the logs, and didn't notice the last line that actually caused the container to stop :)
`level=error msg="migrate: Error 1146: Table 'photoprism.auth_sessions' doesn't exist in auth_sessions (panic)`

I'm trying to run Photoprism with an existing MariaDB 10.5.8 server since it hosts multiple other databases, rather than creating a new MariaDB v10.5.12+ container just for PhotoPrism.  I know, not supported... Anyway, if anyone is still willing to help, any ideas why I might be getting this error? For over a year now I've been working around this problem by just running` image: photoprism/photoprism:220901-bullseye` which is compatible with MariaDB 10.5.8, but would like to upgrade to `photoprism:latest`.  So when I add `PHOTOPRISM_UNSAFE: "true"` to the Docker Compose and try and start the container with `photoprism:latest` I get the error message above migrating the database.  If that is the reason MariaDB 10.5.8 isn't supported, fair game, just thought I'd check if there was some sort of workaround.

Thanks in advance for taking the time...

**Comment by OdinVex** [comment]
It isn't the only table that'll error. `photos` will too. Stick to an Sqlite for now and when your MariaDB gets updated you can dump the Sqlite into a .sql command and then run that in the MariaDB to migrate your data over. That'd be the proper way to do it. But you could buy yourself time...perhaps unwise...by adjusting the Sql and then proceeding:

`/internal/entity/schema/mariadb.sql`:

```
CREATE TABLE `auth_sessions` (
  `id` varbinary(1000) NOT NULL,
...
```
```
CREATE TABLE `photos` (
...
  `photo_path` varbinary(1000) DEFAULT NULL,
...
```

I do not know if this will be problematic. It...*appears* to be working (I exported Sqlite db and imported content adjusting for those records) and you can always dump it and reimport proper when MariaDB gets updated...just remember to adjust the Sql.

### Overt Emphasis

It _can_ work (using right now), but to reiterate a point it is perhaps unwise. Theoretically you should be perfectly fine if you use filenames and paths that are shorter than 745 in length and you should be able to export the database back into a Sql dump and then adjust the table creation for import back into a clean MariaDB database when your installation finally gets upgraded. Keep backups of anything you don't want lost.

**Reply by lastzero** [reply]
Note that this dump is just for documentation purposes. It's neither completely up to date with the very latest changes, may be incomplete, and must not be used to set up or update production environments.

**Reply by OdinVex** [reply]
It should be updated to reflect the current revision, otherwise it's useless documentation edit: (potentially damaging, definitely non-production).

**Reply by lastzero** [reply]
It is often updated and used to create visual documentation, but not for you to use in combination with a version of your choice and without data. PhotoPrism contains everything you need to create and update a schema for you. Also, if you are using an unsupported database version (which contains a major bug), we unfortunately cannot help you, especially not on weekends. Just wanted to be nice and give you a heads up.

**Reply by OdinVex** [reply]
I wasn't suggesting to use it with any specific version of PhotoPrism except the latest, but it does function and for limited libraries it shouldn't be any problem, especially since it can be exported to MariaDB at a later date when they get updated (although that would require dumping and adjusting the initial table creations in the dump for import to MariaDB, adjusted to 'restore' the correct functionality). I also specifically called it (edit for literal text:) "perhaps unwise" to do so, but possible. It's working in a practical application at this very moment and it's significantly faster than Sqlite (Sqlite has such slow database locking to maintain integrity, thus the drive in my testing). In theory it should be fine, provided paths and session ids are kept short. That should be rather simple with non+ editions and most libraries.

**Reply by lastzero** [reply]
If you check our public road map and commit logs, you will find that authentication is under active development right now. You will further find the command to use to reinitialize / reset the sessions table for you:

```
photoprism auth reset
```

Also, please free to contribute if our documentation is not good and up-to-date enough in your opinion. Thank you! ❤️

**Reply by OdinVex** [reply]
It isn't about opinion, it's about development integrity, accuracy (edit: not speaking about developer(s), I'm speaking about the _process of development_, this is not a shot at anyone). If you say the Sql provided as part of the 'current source' is not up-to-date then it is unreliable for anyone to draw conclusions or understanding or use of it. That's just a development issue and I'm only stating just that. I wonder if you can automate the creation of the Sql using a GitHub Action?

**Reply by lastzero** [reply]
As I've said, it's under active development and we will update it with our next stable release as we do not have the resources to update our entire documentation after every single commit. That really should not be too hard to understand?

**Reply by lastzero** [reply]
And if that's so important to you then feel free to write such an action that does the job without cluttering the commit logs and leading to a major increase in the size of our repository.

**Reply by OdinVex** [reply]
>we will update it with our next stable release

I didn't see that anywhere. I've refreshed the paged, check GitHub's e-mails showing each post, nowhere for me to see.

> we do not have the resources to update our entire documentation after every single commit

Of course, no one would expect _documentation_ to always be accurate, but source? It's source. This was from a stable release source, it is reasonable to conclude it should be accurate to _that_ release.

> And if that's so important to you then feel free to write such an action that does the job without cluttering the commit logs and leading to a major increase in the size of our repository.

It should be important to you, as a developer. It was also a suggestion on how to automate the task. A simple action to create a simple Sql would not clutter commit logs or lead to any major increase, it'd be quite minor indeed, you're only adding an Action and creating a single file for use. \*eyeroll\*

> That really should not be too hard to understand?

Hurdur herpderp?

Good luck @Jobsays. :)

**Reply by lastzero** [reply]
Either you contribute or you stop complaining. We specifically use this to create the following, visual diagram which is updated as our time allows and does not get wasted with such pointless discussions:
https://docs.photoprism.app/developer-guide/database/schema/

**Reply by lastzero** [reply]
Oh look, there's a disclaimer as we've already seen this coming:

**Please keep in mind that some information may be outdated or missing, as these docs were automatically generated from the source code.**

**Reply by lastzero** [reply]
![IMG_2855](https://github.com/photoprism/photoprism/assets/301686/c0f0d2fd-9903-4597-abf8-88af57894e64)

You didn't see this in our photoprism-docs repository? No schema documentation updates whatsoever?

**Reply by OdinVex** [reply]
**I wasn't complaining**, you were because I simply pointed out the inaccuracy creates a development issue. **I also did contribute.** The contribution wasn't "try PhotoPrism's paid option" but **it was a contribution** and it _does_ work, **even if not recommended or wise**.

Edit: You're assuming I navigated to your website instead of looking at your code (the ultimate documentation). I have no use for your site, just the code.

Edit: For the sake of poor @Jobsays's post I'm just going to ignore from now on.

Again, good luck @Jobsays. :) You'll find my contribution and a "working" (but again, perhaps unwise and definitely unpaid) solution above. Remember to keep backups.

**Reply by lastzero** [reply]
We do the best we can  with the time available and even added a disclaimer. I'm sorry if that still causes such major issues to you. If you want a dump, simply use the mariadb dump command instead of some static file you find and that nobody told you to use for that purpose.

**Reply by Jobsays** [reply, marked as answer]
Thanks, @OdinVex for your effort to try and help me rescue my unsupported configuration, but happily I'd already worked around it.  Shoulda marked this topic closed when I did, my bad. And @lastzero I totally get that you have bigger fish to fry than helping some guy with an unsupported configuration anyway, no worries :)   The problem I was having was changing my docker compose from:
image: photoprism/photoprism:220901-bullseye
To:
image: photoprism/photoprism:latest 
With MariaDB 10.5.8 which is unsupported. So I finally broke down and updated my docker compose to also create the dedicated MariaDB container for PhotoPrism until I can get my shared server upgraded. Initially I created a MariaDB 10.11 container but then got the error upgrading, "level=error msg="record not found", apparently because the next PhotoPrism version, 221102-jammy, isn't compatible with that MariaDB version, so I recreated the container at MariaDB 10.5.12 (minimum required version), after which I successfully upgraded to photoprism:221102-jammy, imported my old database, and ran a complete rescan of the library as recommended in the release notes. After that I was able to upgrade to photoprism:latest successfully.

Again, thanks for your efforts both!  We can probably mark this issue closed now.

**Reply by OdinVex** [reply]
You are most welcome. Have a lovely day. :)
